### PR TITLE
UI - Validation Updates - QA Feedback

### PR DESCRIPTION
**Pull request by shaangill025**
*Issue:*

- https://github.com/bcgov/entity/issues/23743

*Description of changes:*
Fix this: https://github.com/bcgov/entity/issues/23743#issuecomment-2438878721

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: strr-web/stores/strr.ts [hunk 1]
```diff
@@ -106,22 +118,46 @@ export const propertyManagerSchema = z.object({
...
+    message: 'Preferred Name must not exceed 50 characters',
+    path: ['contact', 'preferredName']
+  }
+).refine(
+  data => !data.contact.extension || data.contact.extension.length <= 15,
+  {
+    message: 'Extension must not exceed 15 characters',
+    path: ['contact', 'extension']
+  }
+)
```
Comment (shaangill025) [inline strr-web/stores/strr.ts:160]: `max` doesn't work with `optional` [zod] and this is the way to implement field limit for optional fields.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-227-lj3xp8me.web.app

**Review by rstens — APPROVED** [review]
LGTM

**Review by dimak1 — APPROVED** [review]

**Pull request merged**